strawberryfields/api/job.py

```diff
@@ -119,14 +121,27 @@ def result(self) -> Result:
             )
         return self._result
 
+    @property
+    def meta(self) -> dict:
+        """Returns a dictionary of metadata on job execution, such as error
+        details.
+
+        Returns:
+            dict[str, str]
+        """
+        return self._meta
+
     def refresh(self):
-        """Refreshes the status of an open or queued job,
+        """Refreshes the status and metadata of an open or queued job,
         along with the job result if the job is newly completed.
         """
         if self._status.is_final:
             self.log.warning("A %s job cannot be refreshed", self._status.value)
             return
-        self._status = JobStatus(self._connection.get_job_status(self.id))
+        job_info = self._connection.get_job(self.id)
+        self._status = JobStatus(job_info.status)
```

Comment:
Logically for me, the `meta` data belongs to the status of the `Job`. At least this attribute will change the similarly to the change in `status` and as far as I understand it, is meant to provide more details about the status of the `Job`.

Reply:
Yes, I agree with the general idea, and also that it would be a relatively big change, so it seems best to tackle that in the future :slightly_smiling_face: